### PR TITLE
docs: refactor ./docs into per-module files

### PR DESCRIPTION
**Pull request by ahdinosaur**
i'm working on [aggregating the pull module ecosystem into a website](https://github.com/pull-stream/pull-stream-docs).

this pull request helps the modules glued to the core `pull-stream` be consistent with rest of pull module ecosystem in having their own markdown file for documentation.

### CODE REVIEW

**Comment by dominictarr** [comment]
it's really annoying to have the .md next to .js with the same name, you be against having this under a `docs` dir? either `docs/{type}/{fn}.md` or `{type}/docs/{fn}.md`?

**Comment by ahdinosaur** [comment]
i can change it to `docs/{type}/{fn}.md` if that's what you'd prefer. :+1: 

just curious, what makes it annoying to you? i've been wondering why `{type}/{fn}.js` instead of `{fn}.js`, possibly related.

**Comment by dominictarr** [comment]
it interferes with tab completion.
`{type}/{fn}.js` was a different reason.
that was because it made it non-breaking change, since `require('pull-stream/sources')` still works, just now it resolves to `pull-stream/sources/index`. My policy on directories is that a directory should represent a distinction. `tests`, `examples`, `node_modules`...
I hate `lib` because it doesn't really mean anything.
basically it means "junk drawer". If you see lib in one of my repos know there is a weird legacy or some junk in there.
`src` is acceptable because it means "to be compiled before use"

**Comment by dominictarr** [comment]
merged into 3.4.2
